backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/IrModuleDeserializer.kt

```diff
@@ -0,0 +1,1133 @@
+        val classType = deserializeIrType(proto.type)
+        /** TODO: [createClassifierSymbolForClassReference] is internal function */
+        return IrClassReferenceImpl(start, end, type, symbol, classType)
+    }
+
+    private fun deserializeCall(proto: KonanIr.IrCall, start: Int, end: Int, type: IrType): IrCall {
+        val symbol = deserializeIrSymbol(proto.symbol) as IrFunctionSymbol
+
+        val superSymbol = if (proto.hasSuper()) {
+            deserializeIrSymbol(proto.`super`) as IrClassSymbol
+        } else null
+
+        val call: IrCall = when (proto.kind) {
+            KonanIr.IrCall.Primitive.NOT_PRIMITIVE ->
+                // TODO: implement the last three args here.
+                IrCallImpl(
+                    start, end, type,
+                    symbol, symbol.descriptor,
+                    proto.memberAccess.typeArguments.typeArgumentCount,
+                    proto.memberAccess.valueArgumentList.size,
+                    null, superSymbol
+                )
+            KonanIr.IrCall.Primitive.NULLARY ->
+                IrNullaryPrimitiveImpl(start, end, type, null, symbol)
+            KonanIr.IrCall.Primitive.UNARY ->
+                IrUnaryPrimitiveImpl(start, end, type, null, symbol)
+            KonanIr.IrCall.Primitive.BINARY ->
+                IrBinaryPrimitiveImpl(start, end, type, null, symbol)
+            else -> TODO("Unexpected primitive IrCall.")
+        }
+        deserializeMemberAccessCommon(call, proto.memberAccess)
+        return call
+    }
+
+    private fun deserializeComposite(proto: KonanIr.IrComposite, start: Int, end: Int, type: IrType): IrComposite {
+        val statements = mutableListOf<IrStatement>()
+        val statementProtos = proto.statementList
+        statementProtos.forEach {
+            statements.add(deserializeStatement(it) as IrStatement)
+        }
+        return IrCompositeImpl(start, end, type, null, statements)
+    }
+
+    private fun deserializeDelegatingConstructorCall(
+        proto: KonanIr.IrDelegatingConstructorCall,
+        start: Int,
+        end: Int
+    ): IrDelegatingConstructorCall {
+        val symbol = deserializeIrSymbol(proto.symbol) as IrConstructorSymbol
+        val call = IrDelegatingConstructorCallImpl(
+            start,
+            end,
+            builtIns.unitType,
+            symbol,
+            symbol.descriptor,
+            proto.memberAccess.typeArguments.typeArgumentCount,
+            proto.memberAccess.valueArgumentList.size
+        )
+
+        deserializeMemberAccessCommon(call, proto.memberAccess)
+        return call
+    }
+
+
+    fun deserializeEnumConstructorCall(
+        proto: KonanIr.IrEnumConstructorCall,
+        start: Int,
+        end: Int,
+        type: IrType
+    ): IrEnumConstructorCall {
+        val symbol = deserializeIrSymbol(proto.symbol) as IrConstructorSymbol
+        return IrEnumConstructorCallImpl(
```

Comment:
Is `deserializeMemberAccessCommon` missing here?

Reply:
done